### PR TITLE
Add unit test for inputWithIcon component

### LINKED ISSUE #96: (SP: 1) Write unit test for "InputWithIcon" component
### Component unit test
Unit test for "**InputWithIcon**" component
Scenaries descriptions:
- [ ] Check if the "**start**" icon that was passed in props should be in the document 
- [ ] If the value of input is not empty it should show "**clear**" icon

[Link to component](https://github.com/Space2Study-UA-1125/frontEnd/blob/develop/src/components/input-with-icon/InputWithIcon.jsx)

Current coverage:
Image

### PR DESCRIPTION
**Pull request by iaremchuk**
Unit test for "InputWithIcon" component
Scenaries descriptions:
 - Check if the "start" icon that was passed in props should be in the document
 -  If the value of input is not empty it should show "clear" icon

Untitled design-104

### CODE REVIEW

**Review by OliaHrytsak — APPROVED** [review]

**Review by VictoriaRodik — APPROVED** [review]

**Review by StasRusnak1 — APPROVED** [review]

**Review by iva-stasia — APPROVED** [review]

File: src/components/input-with-icon/InputWithIcon.jsx [hunk 1]
```diff
@@ -7,9 +7,9 @@ import { styles } from '~/components/input-with-icon/InputWithIcon.styles'
 
 const InputWithIcon = ({ value, onClear, startIcon, sx, ...props }) => {
   return (
-    <Box sx={[styles.root, sx]}>
+    <Box data-testid='inputWithIconContainer' sx={[styles.root, sx]}>
       {startIcon}
-      <InputBase sx={styles.input} value={value} {...props} />
+      <InputBase data-testid='inputWithIconInputBase' sx={styles.input} value={value} {...props} />
```
Comment (vlad-khrychov) [inline src/components/input-with-icon/InputWithIcon.jsx]: Sonar do not like this row,fix it, please

File: tests/unit/components/input-with-icon/InputWithIcon.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        expect(startIcon).toBeInTheDocument()
+    })
+    it('renders inputBase', () => {
+        const inputWithIconInputBase = screen.getByTestId('inputWithIconInputBase')
+        expect(inputWithIconInputBase).toBeInTheDocument()
+    })
+    it('renders clear icon when input value is not empty', () => {
+        const clearIcon = screen.getByTestId('clearIcon')
+        expect(clearIcon).toBeInTheDocument()
+    });
```
Comment (vlad-khrychov) [inline tests/unit/components/input-with-icon/InputWithIcon.spec.jsx]: Let`s add additional test when value is empty.

**Review by vlad-khrychov — CHANGES_REQUESTED** [review]
Maybe you can delete `data-testid` from `InputBase` and add mock for this component?

File: src/components/input-with-icon/InputWithIcon.jsx (hunk 1 above)
Comment (iaremchuk) [reply src/components/input-with-icon/InputWithIcon.jsx]: Thank you for reviewing the tests report and for bringing to my attention that Sonar does not accept the attribute data-testid='inputWithIconInputBase'. I discovered that the InputBase component from MUI has the role 'textbox':

Here are the accessible roles:
  textbox:
  Name "":
  <input
    class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
    type="text"
    value="value"
  />

As a result, I have updated the testing approach from using getByTestId to getByRole to avoid the use of the data-testid='inputWithIconInputBase' attribute.

File: tests/unit/components/input-with-icon/InputWithIcon.spec.jsx (hunk 2 above)
Comment (iaremchuk) [reply tests/unit/components/input-with-icon/InputWithIcon.spec.jsx]: Done, thanks

**Review by holus-bolus — APPROVED** [review]

**Review by vlad-khrychov — APPROVED** [review]
